File: articles/broaden-and-build-theory-positive-psychology/fortran/broaden_build_index.f90
program broaden_build_index
  implicit none
  real :: positive_emotion, negative_emotion, cognitive_flexibility, exploratory_behavior
  real :: affiliative_behavior, social_support, resilience_score, stress_arousal
  real :: contextual_safety, resource_stock, practice_fit
  real :: broadening, resource, recovery, net_adaptation

  positive_emotion = 7.2
  negative_emotion = 3.8
  cognitive_flexibility = 7.1
  exploratory_behavior = 7.0
  affiliative_behavior = 7.2
  social_support = 7.3
  resilience_score = 6.9
  stress_arousal = 3.6
  contextual_safety = 7.2
  resource_stock = 7.1
  practice_fit = 7.3

  broadening = (positive_emotion + cognitive_flexibility + exploratory_behavior + &
      affiliative_behavior + contextual_safety - negative_emotion) / 6.0

  resource = (social_support + resilience_score + contextual_safety + resource_stock + practice_fit) / 5.0

  recovery = (positive_emotion + social_support + contextual_safety + resilience_score - &
      stress_arousal - negative_emotion) / 6.0

  net_adaptation = positive_emotion + cognitive_flexibility + exploratory_behavior + &
      affiliative_behavior + social_support + resilience_score + contextual_safety + &
      resource_stock + practice_fit - negative_emotion - stress_arousal

  print *, "Broadening index:", broadening
  print *, "Resource index:", resource
  print *, "Recovery capacity index:", recovery
  print *, "Net adaptation index:", net_adaptation
end program broaden_build_index

File: articles/broaden-and-build-theory-positive-psychology/go/broaden_build_index.go
package main

import "fmt"

type BroadenBuildIndicators struct {
	PositiveEmotion      float64
	NegativeEmotion      float64
	CognitiveFlexibility float64
	ExploratoryBehavior  float64
	AffiliativeBehavior  float64
	SocialSupport        float64
	ResilienceScore      float64
	StressArousal        float64
	ContextualSafety     float64
	ResourceStock        float64
	PracticeFit          float64
}

func BroadeningIndex(x BroadenBuildIndicators) float64 {
	return (x.PositiveEmotion +
		x.CognitiveFlexibility +
		x.ExploratoryBehavior +
		x.AffiliativeBehavior +
		x.ContextualSafety -
		x.NegativeEmotion) / 6.0
}

func ResourceIndex(x BroadenBuildIndicators) float64 {
	return (x.SocialSupport +
		x.ResilienceScore +
		x.ContextualSafety +
		x.ResourceStock +
		x.PracticeFit) / 5.0
}

func RecoveryCapacityIndex(x BroadenBuildIndicators) float64 {
	return (x.PositiveEmotion +
		x.SocialSupport +
		x.ContextualSafety +
		x.ResilienceScore -
		x.StressArousal -
		x.NegativeEmotion) / 6.0
}

func NetAdaptationIndex(x BroadenBuildIndicators) float64 {
	return x.PositiveEmotion +
		x.CognitiveFlexibility +
		x.ExploratoryBehavior +
		x.AffiliativeBehavior +
		x.SocialSupport +
		x.ResilienceScore +
		x.ContextualSafety +
		x.ResourceStock +
		x.PracticeFit -
		x.NegativeEmotion -
		x.StressArousal
}

func main() {
	example := BroadenBuildIndicators{7.2, 3.8, 7.1, 7.0, 7.2, 7.3, 6.9, 3.6, 7.2, 7.1, 7.3}
	fmt.Printf("Broadening index: %.3f\n", BroadeningIndex(example))
	fmt.Printf("Resource index: %.3f\n", ResourceIndex(example))
	fmt.Printf("Recovery capacity index: %.3f\n", RecoveryCapacityIndex(example))
	fmt.Printf("Net adaptation index: %.3f\n", NetAdaptationIndex(example))
}
